network-api/networkapi/wagtailpages/pagemodels/customblocks/common/base_link_block.py

```diff
@@ -97,6 +81,9 @@ def clean(self, value):
         url_default_values = self.get_default_values()
         url_type = clean_values.get("link_to")
 
+        if not url_type:
+            errors["link_to"] = ErrorList(["Please select a link type"])
```

Comment:
Could we instead use required=True in the model's field? Or do we need the additional flexibility?

Reply:
This is a tricky one. All blocks have `required=True` by default but since `link_to` is a `ChoiceBlock`, it contains a blank choice (the empty string `""`). So, if editors do not select a link type, technically the block still has some data.

If we put `required=True` but remove that validation check, it will allow editors to save without selecting a "valid" type of link:
![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/61277874/5ccff1ab-a305-404f-bafc-5af245817d23)

That's why we have to put that `if not url_type` check, which will trigger an error if editors have not made a selection (the choice is blank):
![image](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/61277874/7739ee01-f3f0-4e46-b133-768b4ae3fb11)

Reply:
Gotcha, thanks @jhonatan-lopes! I just tried that locally with the `BaseLinkBlock` and do see what you mean here.

I messed around with it a little bit further, and it looks like if we remove the required=False from the `LinkBlock` class, it will give us the red * and the required validation even in the case that the user selects the blank option and the additional `url_type` check is removed. Not a big deal, and it's up to you. But if we can, I feel like it'd be a consistent editor experience to have the * and out of the box validation.

![Screenshot 2024-03-13 at 9 37 33 AM](https://github.com/MozillaFoundation/foundation.mozilla.org/assets/3083419/b9e38567-9602-4c5e-9cdd-e4381f03a9d1)

Reply:
I see what you mean. Yes, having the required * is indeed an advantage. I'll add that.

Reply:
You were absolutely right, @robdivincenzo! Removing the `required=False` and doing some new migrations makes it so that the check is redundant. I've fixed that now. Thanks